src/character.cpp

```diff
@@ -4213,10 +4213,6 @@ std::pair<std::string, nc_color> Character::get_hunger_description() const
         } else if( recently_ate && contains >= cap * 3 / 8 ) {
             hunger_string = _( "Full" );
             hunger_color = c_green;
-        } else if( ( stomach.time_since_ate() > 90_minutes && contains < cap / 8 && recently_ate ) ||
-                   ( just_ate && contains > 0_ml && contains < cap * 3 / 8 ) ) {
-            hunger_string = _( "Peckish" );
-            hunger_color = c_dark_gray;
```

Comment:
Probably better to merge the branch condition into the next branch, otherwise people may see "Hungry/Very Hungry" when they would have see "Peckish" before this change.

Reply:
I carefully analyzed that, and that's also why I outlined the testing I did.
As far as I can tell, "Peckish" only ever kicked in instead of no message, so now it should just go to no message as intended.  Also this logic had gotten significantly over-complicated.
If you're hungry and start eating, you should still see hungry until either 15 minutes have passed OR you reach 3 / 8 stomach fullness.
If you're full and passing time as your stomach empties out, it should go from "Sated" to no message, then to Hungry.
If you're hungry and eat a small snack, it should go from hungry to no message and back to hungry, unless the snack is so small that your stomach empties out within 15 minutes, in which case it never budges from hungry.

Reply:
> As far as I can tell, "Peckish" only ever kicked in instead of no message

If `just_ate` is true, "Peckish" may kick in but no message may not. For example, if the stomach content is between 1/8 and 3/8, and time since last ate is larger than 0.25 hrs, then no hunger message is displayed. If at this time the player eats something small (such as a pill of aspirin?), then time since last ate becomes 0 and stomach content is still below 3/8. Before this change "Peckish" would have been displayed, and after this change "Hungry" would be displayed.

Reply:
Going from no message to "peckish" when you eat and going from no message to "hungry" when you eat are both bugs, so restoring the old behaviour isn't a fix.
The only pertinent part of that logic is "just ate", everything else is redundant.  The problem is, if we don't exclude "just ate" from the no message case, then "hungry" will always disappear immediately upon you eating anything, which is most likely going to make people think "eat one thing and then stop" is what the game wants them to do.

I can't see any way to resolve this without overhauling the whole thing, the scenario of "eating a tic-tack when your stomach is almost empty makes the game say you're hungry" is unintended, but it's the least bad alternative I can see, and it's no worse than the status quo.